src/lib/DepositRecordSerializer.js

```diff
@@ -190,6 +228,13 @@ export class DepositRecordSerializer {
       );
     }
     // Remove empty values again because serialization may add some back
-    return this.removeEmptyValues(serializedRecord);
+    serializedRecord = this.removeEmptyValues(serializedRecord);
```

Comment:
I think would be cleaner in some point to ensure that the serializer doesn't set empty values instead of removing them again....